src/AlgebraicGeometry/Schemes/AffineSchemes/Objects/Properties.jl

```diff
@@ -654,6 +650,53 @@ end
 is_smooth(X::AbsSpec{<:Field, <:MPolyRing}) = true
 is_smooth(X::AbsSpec{<:Field, <:MPolyLocRing}) = true
 
+###################################################################
+# Irreducibility and Integrality                                  #
+#    integral iff (irreducible && reduced)                        #
+#    integral = OO(X) is integral domain                          #
+#    irreducible = nilradical of OO(X) is prime                   #
+###################################################################
+@doc Markdown.doc"""
+   is_irreducible(X::AbsSpec)
+
+Return the boolean value whether an affine scheme `X` is irreducible.
+
+"""
+@attr function is_irreducible(X::AbsSpec)
+  !is_empty(X) || return false
+  !get_attribute(X, :is_integral, false) || return true 
+                                           ## integral = irreducible + reduced
+  return (length(minimal_primes(saturated_ideal(modulus(OO(X))))) == 1)
+end
+
+@doc Markdown.doc"""
+   is_integral(X::AbsSpec)
+
+Return the boolean value whether an affine scheme `X` is integral,
+i.e. irreducible and reduced.
+
+"""
 @attr function is_integral(X::AbsSpec)
-  return Oscar._is_integral_domain(OO(X))
+  !is_empty(X) || return false
```

Comment:
for speed only?

Reply:
No, the point was to use the function is_prime, which is officially available, instead of an internal one which calls is_prime anyway.